iroh-bytes/src/get/io.rs

```diff
@@ -0,0 +1,522 @@
+                        .context("final path should have parent")?,
+                )
+                .await?;
+                // Rename temp file, to target name
+                // once this is done, the file is considered complete
+                tokio::fs::rename(data_path, final_path).await?;
+                if let Some(mut outboard_file) = outboard_file.take() {
+                    // not sure if we have to do this
+                    outboard_file.sync().await?;
+                    // delete the outboard file
+                    tokio::fs::remove_file(outboard_path).await?;
+                }
+                curr
+            };
+            pb.finish();
+            next = curr.next();
+        };
+        let stats = finishing.next().await?;
+        tokio::fs::remove_dir_all(temp_dir).await?;
+        pb.finish_and_clear();
+        writer.write(format!(
+            "Transferred {} in {}, {}/s",
+            HumanBytes(stats.bytes_read),
+            HumanDuration(stats.elapsed),
+            HumanBytes((stats.bytes_read as f64 / stats.elapsed.as_secs_f64()) as u64)
+        ));
+
+        Ok(())
+    }
+
+    /// Get into a file or directory
+    async fn get_to_dir(self, writer: &StdWriter, out_dir: PathBuf) -> Result<()> {
+        let single = self.single();
+        let temp_dir = out_dir.join(".iroh-tmp");
+        if single {
+            self.get_to_file_single(writer, out_dir, temp_dir).await
+        } else {
+            self.get_to_dir_multi(writer, out_dir, temp_dir).await
+        }
+    }
+
+    pub async fn get_interactive(self, writer: &StdWriter, out_dir: Option<PathBuf>) -> Result<()> {
+        if let Some(out_dir) = out_dir {
+            self.get_to_dir(writer, out_dir).await
+        } else {
+            self.get_to_stdout(writer).await
+        }
+    }
+
+    /// Get to stdout, no resume possible.
+    async fn get_to_stdout(self, writer: &StdWriter) -> Result<()> {
+        let hash = self.hash();
+        let single = self.single();
+        writer.write(format!("Fetching: {}", Blake3Cid::new(hash)));
+        writer.write(format!("{} Connecting ...", style("[1/3]").bold().dim()));
+        let query = if single {
+            // just get the entire first item
+            RangeSpecSeq::new([get::RangeSet2::all()])
+        } else {
+            // get everything (collection and children)
+            RangeSpecSeq::all()
+        };
+
+        let pb = make_download_pb();
+        let request = GetRequest::new(self.hash(), query).into();
+        let response = match self {
+            GetInteractive::Ticket {
+                ticket,
+                keylog,
+                derp_map,
+            } => get::run_ticket(&ticket, request, keylog, derp_map).await?,
+            GetInteractive::Hash { opts, .. } => get::run(request, opts).await?,
+        };
+        let connected = response.next().await?;
+        // progress!("{} Requesting ...", style("[2/3]").bold().dim());
```

Comment:
Leaving this out makes the CLI output change / not make sense. We have 1/3 & 3/3 but no 2/3 anymore.

Reply:
ahh I missed this 😓 thanks